modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java

```diff
@@ -1223,6 +1222,72 @@ public void testAddObjectActionWithConditionExpression() throws Exception {
 		_objectActionLocalService.deleteObjectAction(objectAction4);
 	}
 
+	@Test
+	public void testAddObjectActionWithLabelMap() throws Exception {
+		Locale locale = LocaleUtil.fromLanguageId(
+			_objectDefinition.getDefaultLanguageId());
+
+		String objectActionName = RandomTestUtil.randomString();
+
+		ObjectAction objectAction1 = _addObjectAction(
+			StringPool.BLANK, null, objectActionName,
+			ObjectActionExecutorConstants.KEY_GROOVY,
+			ObjectActionTriggerConstants.KEY_ON_AFTER_DELETE,
+			new UnicodeProperties(), false);
+
+		Assert.assertEquals(
+			Collections.singletonMap(locale, objectActionName),
+			objectAction1.getLabelMap());
+
+		String label1 = RandomTestUtil.randomString();
+
+		ObjectAction objectAction2 = _addObjectAction(
+			StringPool.BLANK,
+			HashMapBuilder.put(
+				LocaleUtil.BRAZIL, label1
+			).build(),
+			RandomTestUtil.randomString(),
+			ObjectActionExecutorConstants.KEY_GROOVY,
+			ObjectActionTriggerConstants.KEY_ON_AFTER_DELETE,
+			new UnicodeProperties(), false);
+
+		Assert.assertEquals(
+			HashMapBuilder.put(
+				locale, label1
+			).put(
+				LocaleUtil.BRAZIL, label1
+			).build(),
+			objectAction2.getLabelMap());
+
+		String label2 = RandomTestUtil.randomString();
+		objectActionName = RandomTestUtil.randomString();
+
+		ObjectAction objectAction3 = _addObjectAction(
+			StringPool.BLANK,
+			HashMapBuilder.put(
+				LocaleUtil.BRAZIL, label1
+			).put(
+				LocaleUtil.SPAIN, label2
+			).build(),
+			objectActionName, ObjectActionExecutorConstants.KEY_GROOVY,
+			ObjectActionTriggerConstants.KEY_ON_AFTER_DELETE,
+			new UnicodeProperties(), false);
+
+		Assert.assertEquals(
+			HashMapBuilder.put(
+				locale, objectActionName
+			).put(
+				LocaleUtil.BRAZIL, label1
+			).put(
+				LocaleUtil.SPAIN, label2
```

Comment:
Replace `label1` with `"pt_BR objectAction2Label"` and `label2` with `"es_ES objectAction2Label"`